controllers/subscription.go

```diff
@@ -70,30 +72,76 @@ func newSubscription(p api.Pattern) *operatorv1alpha1.Subscription {
 		},
 	}
 
-	if p.Spec.GitOpsConfig.UseCSV {
-		spec.StartingCSV = p.Spec.GitOpsConfig.OperatorCSV
+	if input.Channel != "" {
+		spec.Channel = input.Channel
+	}
+
+	if input.CatalogSource != "" {
+		spec.CatalogSource = input.CatalogSource
+	}
+
+	if input.StartingCSV != "" {
+		spec.StartingCSV = input.StartingCSV
 	}
 
-	if p.Spec.GitOpsConfig.ManualApproval {
-		spec.InstallPlanApproval = operatorv1alpha1.ApprovalManual
+	if spec.InstallPlanApproval != "" {
+		spec.InstallPlanApproval = input.InstallPlanApproval
 	}
 
 	return &operatorv1alpha1.Subscription{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "openshift-gitops-operator",
+			Name:      gitopsSubscriptionName,
 			Namespace: subscriptionNamespace,
 		},
 		Spec: spec,
 	}
 }
 
-func getSubscription(client olmclient.Interface, name, namespace string) (error, *operatorv1alpha1.Subscription) {
+func newSubscriptionFromConfigMap(r kubernetes.Interface) (operatorv1alpha1.Subscription, error) {
+	var newSpec *operatorv1alpha1.Subscription
+
+	// Check if the config map exists and read the config map values
+	if cm, err := r.CoreV1().ConfigMaps(common.OperatorNamespace).Get(context.Background(), common.OperatorConfigFile, metav1.GetOptions{}); err != nil {
+		fmt.Println("Patterns Config Map not found. Using default subscriptions values for OpenShift GitOps.")
```

Comment:
let's use the log subsystem for consistency

Reply:
Actually, I think it'd be nicer to drop this branch of the if entirely and make sure the default logic when the configmap is missing, comes from configValueWithDefault directly. What do you think? That way we can drop newSubscription() entirely

Reply:
@mbaldessari That is a good idea.  Looking into that implementation today 10/9/2023.

Reply:
@mbaldessari agreed, i discussed this approach with @claudiol last week

Reply:
newSubscription() function dropped.